src/main/scala/no/ndla/draftapi/service/StateTransitionRules.scala

```diff
@@ -116,27 +116,33 @@ trait StateTransitionRules {
        USER_TEST                  -> DRAFT,
       (USER_TEST                  -> AWAITING_QUALITY_ASSURANCE) keepStates Set(IMPORTED, PROPOSAL, PUBLISHED) keepCurrentOnTransition,
       (USER_TEST                  -> PUBLISHED)                  keepStates Set(IMPORTED) require DirectPublishRoles withSideEffect publishWithSoftValidation,
+      USER_TEST                   -> ARCHIVED                    require PublishRoles illegalStatuses Set(PUBLISHED),
       (AWAITING_QUALITY_ASSURANCE -> AWAITING_QUALITY_ASSURANCE) keepStates Set(IMPORTED, PROPOSAL, USER_TEST, PUBLISHED),
        AWAITING_QUALITY_ASSURANCE -> DRAFT,
        AWAITING_QUALITY_ASSURANCE -> QUEUED_FOR_LANGUAGE,
       (AWAITING_QUALITY_ASSURANCE -> USER_TEST)                  keepStates Set(IMPORTED, PROPOSAL, PUBLISHED),
       (AWAITING_QUALITY_ASSURANCE -> QUALITY_ASSURED)            keepStates Set(IMPORTED, USER_TEST, PUBLISHED) withSideEffect validateArticleApiArticle,
       (AWAITING_QUALITY_ASSURANCE -> PUBLISHED)                  keepStates Set(IMPORTED) require DirectPublishRoles withSideEffect publishWithSoftValidation,
+      AWAITING_QUALITY_ASSURANCE  -> ARCHIVED                    require PublishRoles illegalStatuses Set(PUBLISHED),
        QUALITY_ASSURED            -> QUALITY_ASSURED,
        QUALITY_ASSURED            -> DRAFT,
       (QUALITY_ASSURED            -> QUEUED_FOR_PUBLISHING)      keepStates Set(IMPORTED, USER_TEST, PUBLISHED) require PublishRoles withSideEffect validateArticleApiArticle keepCurrentOnTransition,
       (QUALITY_ASSURED            -> PUBLISHED)                  keepStates Set(IMPORTED) require DirectPublishRoles withSideEffect publishArticle,
+      QUALITY_ASSURED             -> ARCHIVED                    require PublishRoles illegalStatuses Set(PUBLISHED),
        QUEUED_FOR_PUBLISHING      -> QUEUED_FOR_PUBLISHING       withSideEffect validateArticleApiArticle,
       (QUEUED_FOR_PUBLISHING      -> PUBLISHED)                  keepStates Set(IMPORTED) require DirectPublishRoles withSideEffect publishArticle,
        QUEUED_FOR_PUBLISHING      -> DRAFT,
+      QUEUED_FOR_PUBLISHING       -> ARCHIVED                    require PublishRoles illegalStatuses Set(PUBLISHED),
       (PUBLISHED                  -> DRAFT)                      keepCurrentOnTransition,
       (PUBLISHED                  -> PROPOSAL)                   keepCurrentOnTransition,
       (PUBLISHED                  -> AWAITING_UNPUBLISHING)      require PublishRoles withSideEffect checkIfArticleIsUsedInLearningStep keepCurrentOnTransition,
       (PUBLISHED                  -> UNPUBLISHED)                keepStates Set(IMPORTED) require DirectPublishRoles withSideEffect checkIfArticleIsUsedInLearningStep withSideEffect unpublishArticle,
+      PUBLISHED                   -> ARCHIVED                    require PublishRoles illegalStatuses Set(PUBLISHED),
```

Comment:
For denne overganen må du ha med avpublisering.
```suggestion
      PUBLISHED                   -> ARCHIVED                    require PublishRoles illegalStatuses Set(PUBLISHED) withSideEffect checkIfArticleIsUsedInLearningStep  withSideEffect unpublishArticle,
```

Reply:
Jeg tror `withSideEffect` bare bytter ut side-effekten i transitionen, så vi må nok enten lage en side-effekt som har begge funksjonskallene.
Evt skrive om `StateTransition` så den har en liste med side-effekter eller lignende.

(Ser forsåvidt at det finnes transitions som kaller `withSideEffect` flere ganger. Det er en bug, og bare den siste vil bli kjørt, så vi må fikse de også)